### PR TITLE
mpsl: documentation: Added documentation for mpsl

### PR DESCRIPTION
**Pull request by sofu1**
Added documentation for how to integrate MPSL with applications,
about thread safety, interrupts and priorities.

Signed-off-by: Solveig Fure <Solveig.Fure@nordicsemi.no>

### CODE REVIEW

**Comment by sofu1** [comment]
@b-gent @frederik-vestre-nordic

**Comment by sofu1** [comment]
I'm not sure if these changes should be in the PR that we make for the next mpsl release, or in a separate one like this?

File: mpsl/doc/external_mpsl.rst [hunk 1]
```diff
@@ -0,0 +1,61 @@
+.. _external_mpsl:
```
Comment (sofu1) [inline mpsl/doc/external_mpsl.rst]: We will rename this file. Probably to public_mpsl.rst.

**Comment by b-gent** [comment]
I added some changes but I'm still working on getting the format right. I'll push another commit tonight.

File: mpsl/doc/external_mpsl.rst [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+Library internals
+#################
+
+Before using the Multi-Protocol Service Layer (MPSL) library, make sure you are familiar with its design.
```
Comment (b-gent) [inline mpsl/doc/external_mpsl.rst]: We should really spell that 'Multiprotocol Service Layer' according to our rules

File: mpsl/doc/external_mpsl.rst [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+ * POWER_CLOCK interrupt
+
+   The interrupt must be enabled for clock calibration to work, and it is up to the application to forward any clock related events to :cpp:func:`MPSL_IRQ_CLOCK_Handler` in lower priority.
+   Irrelevant events are ignored, so the application is free to forward all events for the POWER_CLOCK interrupt.
+
+
+ * ``low_prio_irq`` interrupt
+
+   Low priority work is signaled by MPSL by pending the IRQ specified in the ``low_prio_irq`` argument to :cpp:func:`mpsl_init`.
+   When this interrupt is triggered, :cpp:func:`mpsl_low_priority_process` should be called within reasonable time.
```
Comment (b-gent) [inline mpsl/doc/external_mpsl.rst]: 'reasonable time' should be a bit more specific :)

**Comment by b-gent** [comment]
LGTM

File: mpsl/doc/external_mpsl.rst [hunk 4]
```diff
@@ -1,61 +1,65 @@
 .. _external_mpsl:
 
-MPSL
-####
+Library internals
+#################
 
-
-.. toctree::
```
Comment (sofu1) [inline mpsl/doc/external_mpsl.rst]: Why did you remove the toc?

Comment (b-gent) [reply mpsl/doc/external_mpsl.rst]: It didn't have any effect when put here. The toctree element is in the parent document (in README.rst) and it pulls this document as a child document under it.

I'm assuming you wanted to use this toctree to show the outline of the sections in this document. I'm not sure if this can be done in this way (possibly yes), but we don't do this in NCS docset, even in much longer documents.

Comment (sofu1) [reply mpsl/doc/external_mpsl.rst]: Ok, now I understand. No, it's not necessary.

File: mpsl/doc/external_mpsl.rst [hunk 5]
```diff
@@ -0,0 +1,77 @@
+.. _external_mpsl:
+
+Library internals
+#################
+
+Before using the Multi-Protocol Service Layer (MPSL) library, make sure you are familiar with its design.
+
+Peripherals
+===========
```
Comment (b-gent) [inline mpsl/doc/external_mpsl.rst]:
```suggestion
***********
```
(to have it on the same level as 'Thread and Interrupt Safety')

File: mpsl/doc/external_mpsl.rst [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+The following peripherals are owned by MPSL and must not be accessed directly by the application:
+
+-  RTC0
```
Comment (b-gent) [inline mpsl/doc/external_mpsl.rst]:
```suggestion
* RTC0
```

File: mpsl/doc/external_mpsl.rst [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+-  TIMER0
```
Comment (b-gent) [inline mpsl/doc/external_mpsl.rst]:
```suggestion
* TIMER0
```

File: mpsl/doc/external_mpsl.rst [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+-  RADIO
```
Comment (b-gent) [inline mpsl/doc/external_mpsl.rst]:
```suggestion
* RADIO
```

File: mpsl/doc/external_mpsl.rst [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+-  CLOCK
```
Comment (b-gent) [inline mpsl/doc/external_mpsl.rst]:
```suggestion
* CLOCK
```

File: mpsl/doc/external_mpsl.rst [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+-  TEMP
```
Comment (b-gent) [inline mpsl/doc/external_mpsl.rst]:
```suggestion
* TEMP
```

**Comment by sofu1** [comment]
These changes will be pushed to nrfxlib in a PR with the next release of mpsl, so I'm closing this PR.
